messaging/src/main/java/org/axonframework/updates/UpdateCheckerHttpClient.java

```diff
@@ -0,0 +1,100 @@
+import org.axonframework.updates.api.UpdateCheckResponse;
+import org.axonframework.updates.configuration.UsagePropertyProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.management.ManagementFactory;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.Optional;
+
+
+/**
+ * Client for checking for updates and sending anonymous usage data to the AxonIQ servers. This client uses the
+ * {@link UsagePropertyProvider} to determine the URL to send the data to.
+ *
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+@Internal
+public class UpdateCheckerHttpClient {
+
+    private final static Logger logger = LoggerFactory.getLogger(UpdateCheckerHttpClient.class);
+
+    private final HttpClient client;
+    private final UsagePropertyProvider userProperties;
+
+    /**
+     * Creates a new {@code UpdateCheckerHttpClient} with the given {@link UsagePropertyProvider}. The client will use
+     * the properties to determine the URL to send the usage data to.
+     *
+     * @param userProperties The {@link UsagePropertyProvider} to use for retrieving the URL and other properties.
+     */
+    public UpdateCheckerHttpClient(UsagePropertyProvider userProperties) {
+        this.userProperties = userProperties;
+        this.client = HttpClient.newBuilder()
+                                .followRedirects(HttpClient.Redirect.ALWAYS)
+                                .build();
+    }
+
+    /**
+     * Sends a usage request to the AxonIQ servers. If {@code firstRequest} is true, it will send a POST request,
+     * otherwise it will send a PUT request.
+     *
+     * @param updateCheckRequest The {@link UpdateCheckRequest} to send.
+     * @param firstRequest Whether this is the first request or not.
+     * @return An {@link Optional} containing the {@link UpdateCheckResponse} if the request was successful, or empty if it
+     * failed.
+     */
+    public Optional<UpdateCheckResponse> sendRequest(UpdateCheckRequest updateCheckRequest, boolean firstRequest) {
+        String url = userProperties.getUrl() + "?" + updateCheckRequest.toQueryString();
+
+        try {
+            logger.debug("Reporting anonymous usage data to AxonIQ servers at: {}", url);
+            HttpRequest request = HttpRequest
+                    .newBuilder()
+                    .uri(URI.create(url))
+                    .timeout(Duration.ofSeconds(5))
+                    .headers("User-Agent", updateCheckRequest.toUserAgent())
+                    .headers("X-Machine-Id", updateCheckRequest.machineId())
+                    .headers("X-Instance-Id", updateCheckRequest.instanceId())
+                    .headers("X-Uptime", String.valueOf(ManagementFactory.getRuntimeMXBean().getUptime()))
+                    .headers("X-First-Run", firstRequest ? "true" : "false")
+                    .GET().build();
```

Comment:
Could you tell me why have we changed it to GET?

Reply:
Yes, to emphasize the fact that we retrieve data (updates and vulnerabilities).